tagstore/tagstore_rest/v1/views/entities.py

```diff
@@ -62,5 +67,69 @@ def tags(self, request, entity_type=None, pk=None):
             queryset = entity.tags.all()
 
         extracted = [self._convert(tag) for tag in queryset]
-        serializer = EntityTagSerializer({'tags': extracted})
+        return EntityTagSerializer({'tags': extracted})
+
+    def tags(self, request, entity_type=None, pk=None):
+        '''
+        Get a list of all tags belonging to an entity.
+
+        These can be optionally filtered by `taxonomy` params.
+        '''
+        entity = get_object_or_404(Entity, external_id=pk, entity_type=entity_type)
+        taxonomies = request.GET.get('taxonomies', '').split(',')
+
+        serializer = self.serialize_tags(entity, taxonomies)
         return Response(serializer.data)
+
+    def update_tags(self, request, entity_type=None, pk=None):
+        '''
+        Update tags belonging to an entity.
+        '''
+        tagstore = DjangoTagstore()
+
+        entity_obj = get_object_or_404(Entity, external_id=pk, entity_type=entity_type)
+        entity = EntityId(external_id=pk, entity_type=entity_type)
+        freeform_obj, _ = Taxonomy.objects.get_or_create(name='FreeForm')
+        freeform = TaxonomyId(freeform_obj.id)
+
+        if not request.data.get('tags', ''):
+            return Response(status=204)
+
+        for tag in request.data['tags']:
+            # a single free form tag
+            if isinstance(tag, str):
+                try:
+                    _tag = tagstore.get_tag_in_taxonomy(tag, freeform)
+                    if not _tag:
+                        _tag = tagstore.add_tag_to_taxonomy(tag, freeform)
+                    tagstore.add_tag_to(_tag, entity)
+                except ValueError:
+                    return Response(status=500)
```

Comment:
`Response(status=500)` will just return a blank document and an HTTP 500 code, which can make debugging hard. It's better to just let the `ValueError` exception propagate up (which will result in a 500 error response, but this time with a clear message), or to log the exception (`logger.exception(...)`) plus return a 500 response including an error message like `{"status": "error", "reason": f"Unable to add tag {tag} to taxonomy {taxonomy.name}"}`